bin/build.sh

```diff
@@ -1,73 +1,81 @@
-      sed -E "s#DOCUMENT_TITLE#$document_title#g" "$edudoc_header" |
-      sed -E "s#WCA_LOGO_PATH#$logo#g" |
-      sed -E "s#DATE#$compile_date#g" > "$header_html"
+      sed -E "s#\{document_title\}#$document_title#g" "$edudoc_header" |
+      sed -E "s#\{wca_logo_path\}#$logo#g" |
+      sed -E "s#\{date\}#$compile_date#g" |
+      echo -e "$(cat /dev/stdin)\n\n$(cat $file)" > "$file"
     fi
 
-    # Markdown -> HTML
-    pandoc -s --from markdown --to html5 --css=$2 --metadata pagetitle="$document_title" "$file" -o "$html_name"
+    # Markdown => HTML + remove unsupported styles generated by pandoc
+    pandoc -s --from markdown --to html5 --css "$main_stylesheet" --css "$custom_stylesheet" --metadata pagetitle="$document_title" "$file" |
+    sed -E "s#gap: min\(4vw, 1\.5em\);##" |
+    sed -E "s#overflow-x: auto;##" > "$html_path"
 
     # HTML -> PDF
-    if [ $1 = "documents" ]; then
-      wkhtmltopdf --encoding 'utf-8' -T 15mm -B 15mm -R 15mm -L 15mm --quiet "$html_name" "$pdf_name"
-    elif [ $1 = "edudoc" ]; then
-      wkhtmltopdf --encoding 'utf-8' -T 15mm -B 15mm -R 15mm -L 15mm --header-html "$header_html" --footer-center "[page]" --quiet "$html_name" "$pdf_name"
+    if [ "$1" = "documents" ]; then
+      # --presentational-hints makes things like the start attribute of ol elements work,
+      # which is required for adding more content between two consecutive ordered list items
+      weasyprint --presentational-hints --encoding 'utf-8' "$html_path" "$pdf_path"
+    elif [ "$1" = "edudoc" ]; then
+      weasyprint --presentational-hints --encoding 'utf-8' "$html_path" "$pdf_path"
     fi
 
-    # Deliberately NOT using the pipe | operator between pandoc and wkhtml because wkhtml tries to resolve image links
-    # based on the location of its input file. If piping is used, wkhtml assumes /tmp as location,
-    # and links like "./dummy-image.png" become "/tmp/dummy-image.png" instead of "edudoc/$project/dummy-image.png"
+    # Remove the Markdown and HTML files
+    rm "$file" "$html_path"
   done
 }
 
-# Remove potentially cached PDFs from the last build run and make an empty build folder
-rm -rf build
-mkdir build
+# Create the build directory if it didn't exist in the first place
+mkdir -p build
 
-convert_to_pdf documents "$stylesheet"
-convert_to_pdf edudoc "$edudoc_stylesheet"
+# Use the DIRECTORY_TO_BUILD environment variable or the passed argument as the only directory
+# that will have its documents built. If neither is set, fall back to building all directories.
+if [ -n "$DIRECTORY_TO_BUILD" ]; then
+  convert_to_pdf "$DIRECTORY_TO_BUILD"
+elif [ -n "$1" ]; then
+  convert_to_pdf "$1"
+else
+  convert_to_pdf documents
+  convert_to_pdf edudoc
+fi
 
-# Remove all non-PDF files and empty folders from build
+# Remove all non-PDF files and empty directories from build
 find build/ -type f -not -name "*.pdf" -delete
 find build/ -type d -empty -delete
+
+echo -e "\nBuild finished!"
```

Comment:
If you need to echo a line break at the end of the string, something with your previous output is wrong. Never manually encode line breaks in script files, previous `echo` prints always terminate with a new line.

If some `weasyprint` commands don't terminate with a new line and you really really need this extra line break, then issue two `echo` statements with the first `echo`ing an empty string. (And probably raise an issue against WeasyPdf, terminal output commands should always be terminated with line breaks for readability.)

Reply:
Nono, it totally does have a line break before anyways, this just adds another line break to visually separate this line from the stuff above it.

Reply:
If you feel strongly about it, I can change this to two echo statements

Reply:
Hm, in that case I guess it's fine.